geopandas/io/sql.py

```diff
@@ -98,3 +101,227 @@ def load_geom_text(x):
+    https://geoalchemy-2.readthedocs.io/en/latest/types.html#geoalchemy2.types._GISType
+
+    Following rules apply:
+     - if geometries all share the same geometry-type,
+       geometries are inserted with the given GeometryType with following types:
+        - Point, LineString, Polygon, MultiPoint, MultiLineString, MultiPolygon,
+          GeometryCollection.
+        - LinearRing geometries will be converted into LineString -objects.
+     - in all other cases, geometries will be inserted with type GEOMETRY:
+        - a mix of Polygons and MultiPolygons in GeoSeries
+        - a mix of Points and LineStrings in GeoSeries
+        - geometry is of type GeometryCollection,
+          such as GeometryCollection([Point, LineStrings])
+     """
+    geom_types = list(gdf.geometry.geom_type.unique())
+    has_curve = False
+
+    for gt in geom_types:
+        if gt is None:
+            continue
+        elif "LinearRing" in gt:
+            has_curve = True
+
+    if len(geom_types) == 1:
+        if has_curve:
+            target_geom_type = "LINESTRING"
+        else:
+            if geom_types[0] is None:
+                raise ValueError("No valid geometries in the data.")
+            else:
+                target_geom_type = geom_types[0].upper()
+    else:
+        target_geom_type = "GEOMETRY"
+
+    return target_geom_type, has_curve
+
+
+def _get_srid_from_crs(gdf):
+    """
+    Get EPSG code from CRS if available. If not, return -1.
+    """
+
+    # Use geoalchemy2 default for srid
+    # Note: undefined srid in PostGIS is 0
+    srid = -1
+    warning_msg = (
+        "Could not parse CRS from the GeoDataFrame. "
+        + "Inserting data without defined CRS.",
+    )
+    if gdf.crs is not None:
+        try:
+            srid = gdf.crs.to_epsg(min_confidence=25)
+            if srid is None:
+                srid = -1
+                warnings.warn(warning_msg, UserWarning, stacklevel=2)
+        except Exception:
+            warnings.warn(
+                warning_msg, UserWarning, stacklevel=2,
+            )
+    return srid
+
+
+def _convert_linearring_to_linestring(gdf, geom_name):
+    from shapely.geometry import LineString
+
+    # Todo: Use Pygeos function once it's implemented:
+    #  https://github.com/pygeos/pygeos/issues/76
+
+    mask = gdf.geom_type == "LinearRing"
+    gdf.loc[mask, geom_name] = gdf.loc[mask, geom_name].apply(
+        lambda geom: LineString(geom)
+    )
+    return gdf
+
+
+def _convert_to_ewkb(gdf, geom_name, srid):
+    """Convert geometries to ewkb. """
+    if compat.USE_PYGEOS:
+        from pygeos import from_shapely, set_srid, to_wkb
+
+        geoms = to_wkb(
+            set_srid(from_shapely(gdf[geom_name].values), srid=srid),
+            hex=True,
+            include_srid=True,
+        )
+
+    else:
+        from shapely.wkb import dumps
+
+        geoms = [dumps(geom, srid=srid) for geom in gdf[geom_name]]
```

Comment:
```suggestion
        geoms = [dumps(geom, srid=srid, hex=True) for geom in gdf[geom_name]]
```

this is causing the failures on travis

Reply:
Oops, true. Just sent a fix.